pkg/kratos/service.go

```diff
@@ -159,6 +280,29 @@ func (s *Service) UpdateLoginFlow(
 	return &returnToResp, resp.Cookies(), nil
 }
 
+func (s *Service) UpdateSettingsFlow(
+	ctx context.Context, flow string, body kClient.UpdateSettingsFlowBody, cookies []*http.Cookie,
+) (*kClient.SettingsFlow, []*http.Cookie, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.FrontendApi.UpdateSettingsFlow")
+	defer span.End()
+
+	settingsFlow, resp, err := s.kratos.FrontendApi().
+		UpdateSettingsFlow(ctx).
+		Flow(flow).
+		UpdateSettingsFlowBody(body).
+		Cookie(cookiesToString(cookies)).
+		Execute()
+
+	if err != nil && resp.StatusCode != 200 {
```

Comment:
```suggestion
	if err != nil || resp.StatusCode != 200 {
```

do we need the check on the `resp.StatusCode` at all?wouldn't err be enough?

Reply:
it may happen that kratos returns an "error" of info type on 200 response, in that case we want to ignore it